demos/coreMQTT/mqtt_demo_serializer.c

```diff
@@ -541,39 +544,34 @@ int RunCoreMqttSerializerDemo( bool awsIotMqttMode,
 }
 /*-----------------------------------------------------------*/
 
-static void prvGracefulShutDown( Socket_t xSocket )
+static BaseType_t prvGracefulShutDown( Socket_t xSocket )
 {
-    uint8_t ucDummy[ 20 ];
-    const TickType_t xShortDelay = pdMS_TO_TICKS( 250 );
-    BaseType_t xShutdownLoopCount = 0;
+    BaseType_t xStatus = pdFAIL;
+    int32_t lSocketStatus = ( int32_t ) SOCKETS_ERROR_NONE;
 
-    if( xSocket != ( Socket_t ) 0 )
-    {
-        if( xSocket != SOCKETS_INVALID_SOCKET )
-        {
-            /* Initiate graceful shutdown. */
-            SOCKETS_Shutdown( xSocket, SOCKETS_SHUT_RDWR );
-
-            /* Wait for the socket to disconnect gracefully (indicated by FreeRTOS_recv()
-             * returning a FREERTOS_EINVAL error) before closing the socket. */
-            while( SOCKETS_Recv( xSocket, ucDummy, sizeof( ucDummy ), 0 ) >= 0 )
-            {
-                /* Wait for shutdown to complete.  If a receive block time is used then
-                 * this delay will not be necessary as FreeRTOS_recv() will place the RTOS task
-                 * into the Blocked state anyway. */
-                vTaskDelay( xShortDelay );
+    /* Call Secure Sockets shutdown function to close connection. */
+    lSocketStatus = SOCKETS_Shutdown( xSocket, SOCKETS_SHUT_RDWR );
 
-                /* Limit the number of FreeRTOS_recv loops to avoid infinite loop. */
-                if( ++xShutdownLoopCount >= mqttexampleMAX_SOCKET_SHUTDOWN_LOOPS )
-                {
-                    break;
-                }
-            }
+    if( lSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+    {
+        LogError( ( "Failed to close connection: SOCKETS_Shutdown call failed. %d", lSocketStatus ) );
```

Comment:
Should this say `SocketStatus %d` like at the log at the bottom?

Reply:
Yes. Thanks